scalafmt-core/shared/src/main/scala/org/scalafmt/util/TokenClasses.scala

```diff
@@ -4,6 +4,16 @@ import scala.meta.internal.classifiers.classifier
 import scala.meta.tokens.Token
 import scala.meta.tokens.Token._
 
+@classifier
+trait ExtensionKeyword
+object ExtensionKeyword {
+  def unapply(token: Token): Boolean = token match {
+    case _: KwExtension => true
+    case Ident("extension") => true
```

Comment:
So `extension` is still a soft keyword, but it was confirmed that is should be changed as it's impossible distinguish. I reopened the issue in Dotty, since it was closed and plainly ignored https://github.com/lampepfl/dotty/issues/10076

In case it becomes a keyword this case should not be needed and currently scalameta always parses extension as a keyword.

Reply:
I cannot promise that it will be enough for the future, since there is still changes being made, but I am 90 % sure. There is a number of cases that extension as a soft keyword will not just work like:
```
extension(a : Int){
  def b = a
}
```

This can be an extension or invocation of `def extension(a : Int)( c : => Unit)`

Reply:
if we end up keeping this logic, i think that
- this commit needs to be rebased on the `master`
- this trait moved inside @MichalBednarz's new `SoftKeywordClasses` type
- `true` on this line replaced with `dialect.allowExtensionMethods`.

Reply:
So actually, it will be a soft keyword :/ I was told otherwise and now again they changed their minds. Sorry about that, I will fix it in the parser.